resteasy-core/src/main/java/org/jboss/resteasy/core/registry/MatchCache.java

```diff
@@ -26,7 +26,11 @@ public Key(final HttpRequest request, final int start) {
             this.path = start == 0 ? matchingPath : matchingPath.substring(start);
             this.start = start;
             this.method = request.getHttpMethod();
-            this.contentType = request.getHttpHeaders().getMediaType();
+            if (request.getHttpHeaders().getMediaType().getParameters().containsKey("boundary")) {
+               this.contentType = new MediaType(request.getHttpHeaders().getMediaType().getType(), request.getHttpHeaders().getMediaType().getSubtype());
```

Comment:
MediaType.equals() checks the parameters, so I question removing **all** of the parameters when "boundary" is present. Actually, I guess, in principle, that it's possible to use the boundary parameter in a @Consumes annotation, but it doesn't seem likely. So it seems reasonable to me to remove only the "boundary" parameter and leave the others in place.

Reply:
@ronsigal  @patriot1burke The "boundary" parameter check and remove all parameters still can't stop the cache to be invalidated by GET request with ContentType=text/html;version=1~2500.
I think we can directly disable this cache when the request ConteType contains parameter "boundary" or other items to make it less vulnerable. I already sent my change. Please review again.

Reply:
I see - that's the change in RootNode which uses the cache only when there are no parameters. That seems like a reasonable heuristic.